concent_api/concent_api/middleware.py

```diff
@@ -39,3 +50,80 @@ def __call__(self, request):
         response = self.get_response(request)
         response['Concent-Version'] = self._concent_version
         return response
+
+
+def should_use_html(request):
+    """
+    Function inspects request headers and determines is response should be in HTML format.
+    1) By default it HTML should not be used
```

Comment:
HTML/JSON should be used only if it's present in `Accept` or the `Accept` header is missing. Otherwise HTTP 406 should be returned to the client.

Reply:
@cameel , are you sure?
1) It is overcomplicating thing from my PoV. What is the benefit here? Do you expect client to handle HTTP406?
2) 
> The HyperText Transfer Protocol (HTTP) `406 Not Acceptable` client error response code indicates that a response matching the list of acceptable values defined in `Accept-Charset` and `Accept-Language` cannot be served.

So in my opinion we would be using `HTTP406` not according to its purpose.

Reply:
> It is overcomplicating thing from my PoV. What is the benefit here? Do you expect client to handle HTTP406?

If you're implementing a client that can process XML and you tell the server explicitly that you want XML and nothing else, you want it to either give you XML or give you an error. Just giving it JSON is not helpful (even though it's allowed by the spec). If the client did not care about the format, then why didn't it just skip `Accept` header or add `*/*` to it?

I mean, it's not like we'll be running into this case constantly but for me the default thing to do is to be strict about input because it makes it easier to spot your mistakes. I think that the opposite requires justification. It would be a bit like just ignoring invalid input. The difference between returning this error and not doing it is a single `if` so I don't see any reason not to do it.

> So in my opinion we would be using HTTP406 not according to its purpose.

I've seen this on MDN too but the RFC says something different. It lists `Accept` too.

https://tools.ietf.org/html/rfc7231#section-6.5.6
> The 406 (Not Acceptable) status code indicates that the target
>    resource does not have a current representation that would be
>    acceptable to the user agent, according to the proactive negotiation
>    header fields received in the request (Section 5.3), and the server
>    is unwilling to supply a default representation.

https://tools.ietf.org/html/rfc7231#section-5.3
> ```
>    +-------------------+---------------+
>    | Header Field Name | Defined in... |
>    +-------------------+---------------+
>    | Accept            | Section 5.3.2 |
>    | Accept-Charset    | Section 5.3.3 |
>    | Accept-Encoding   | Section 5.3.4 |
>    | Accept-Language   | Section 5.3.5 |
>    +-------------------+---------------+
> ```